commit 9c1b0f9c83d5cbf123b9eaef13d5e91bb57a4570 (refs/changes/48/2348/1)
Author: Tim Kordas <tim.kordas@joyent.com>
Date:   2017-08-08T16:04:58-07:00 (2 years, 2 months ago)
    
    MANTA-3311: check that lockfile creation was successful (and that it is ours)

diff --git a/bin/mako_gc.sh b/bin/mako_gc.sh
index 3e4dfa6..bd8312d 100755
--- a/bin/mako_gc.sh
+++ b/bin/mako_gc.sh
@@ -196,7 +196,15 @@ if [[ -n "$LAST_PID" ]]; then
     fi
 fi
 
+# Save our noclobber setting
+RESTORE=$(shopt -op | grep noclobber)
 echo -n $PID >$PID_FILE
+if [[ $? -ne 0 ]]
+then
+    fatal "Failed to acquire lockfile"
+fi
+# Restore our saved noclobber setting
+eval $RESTORE
 
 # Ok, we're good to start gc
 log "starting gc"
